lib/ecto/migration/runner.ex

```diff
@@ -260,6 +261,19 @@ defmodule Ecto.Migration.Runner do
   defp command({:drop, %Constraint{} = constraint}),
     do: "drop constraint #{constraint.name} from table #{quote_name(constraint.prefix, constraint.table)}"
 
+  defp command({:create, %View{name: nil, query: nil}}),
+    do: raise ArgumentError, "a view must have a name and a query"
```

Comment:
There is no need for those checks. If the name or query is nil, you should check when you create the `View` struct in Ecto.Migration. From there on, you can assume both name and query are valid.